x/foundation/keeper/censorship.go

```diff
@@ -6,6 +6,72 @@ import (
 	"github.com/line/lbm-sdk/x/foundation"
 )
 
+func (k Keeper) GetCensorship(ctx sdk.Context, msgTypeURL string) (*foundation.Censorship, error) {
+	store := ctx.KVStore(k.storeKey)
+	key := censorshipKey(msgTypeURL)
+	bz := store.Get(key)
+	if bz == nil {
+		return nil, sdkerrors.ErrNotFound.Wrap("censorship not found")
+	}
+
+	var censorship foundation.Censorship
+	k.cdc.MustUnmarshal(bz, &censorship)
+
+	return &censorship, nil
+}
+
+func (k Keeper) UpdateCensorship(ctx sdk.Context, censorship foundation.Censorship) error {
+	url := censorship.MsgTypeUrl
+
+	oldCensorship, err := k.GetCensorship(ctx, url)
+	if err != nil {
+		return err
+	}
+
+	newAuthority := censorship.Authority
+	oldAuthority := oldCensorship.Authority
+	if newAuthority >= oldAuthority {
```

Comment:
Would this need the same case ?

Reply:
So, you cannot raise the authority, like x/gov -> x/foundation.

Reply:
> Do you mean we need to use `MsgUpdateCencorship` if you want to move the auth. But the autht to want to change should be lower the enum number. Is it right?

So, you cannot raise the authority, like x/gov -> x/foundation.

Reply:
> Would this need the same case ?

The equality means, the msg has no changes, so it rejects.